tests/app/celery/test_service_callback_tasks.py

```diff
@@ -444,3 +449,31 @@ def test_does_not_send_request_when_inbound_sms_does_not_exist(self, notify_api,
             send_inbound_sms_to_service(inbound_sms_id=uuid.uuid4(), service_id=sample_service.id)
 
         assert mock_send.call_count == 0
+
+
+def test_create_delivery_status_callback_provider_payload(sample_notification):
```

Comment:
You should parametrize "payload".  Then something like:
```
if payload:
    assert decrypted_message.get("provider_payload") == payload
else:
    "provider_payload" not in decrypted_message
```

Reply:
Yes, please either parameterize or break these out into distinct tests. You are testing 3 different things in a single unit test the way you have it now, and this is bad practice. 
A unit test should test a distinct thing. A parametrized unit test does just that, or use three different tests here.